src/query.ts

```diff
@@ -256,8 +262,27 @@ export function stringifyEncodingQueryFieldDef(encQ: EncodingQuery): string {
     fn = SHORT_ENUM_SPEC + '';
   }
 
+  if (encQ.scale && !isEnumSpec(encQ.scale)) {
+
+      if (encQ.scale && !isEnumSpec(encQ.scale)) {
+      var scaleParams = {};
+
+      if (encQ.scale['type']) {
```

Comment:
Note for future PRs -- instead of adding them one by one, you can use a loop, or preferrably `reduce` with array of supported properties under scale.